src/Governance.sol

```diff
@@ -220,27 +166,39 @@ contract Governance is Multicall, UserProxyFactory, ReentrancyGuard, Ownable, IG
     }
 
     /// @inheritdoc IGovernance
-    function depositLQTY(uint88 _lqtyAmount) external nonReentrant {
-        UserProxy userProxy = _updateUserTimestamp(_lqtyAmount);
+    function depositLQTY(uint256 _lqtyAmount) external nonReentrant {
+        UserProxy userProxy = _increaseUserVoteTrackers(_lqtyAmount);
         userProxy.stake(_lqtyAmount, msg.sender);
     }
 
     /// @inheritdoc IGovernance
-    function depositLQTYViaPermit(uint88 _lqtyAmount, PermitParams calldata _permitParams) external nonReentrant {
-        UserProxy userProxy = _updateUserTimestamp(_lqtyAmount);
+    function depositLQTYViaPermit(uint256 _lqtyAmount, PermitParams calldata _permitParams) external nonReentrant {
+        UserProxy userProxy = _increaseUserVoteTrackers(_lqtyAmount);
         userProxy.stakeViaPermit(_lqtyAmount, msg.sender, _permitParams);
     }
 
     /// @inheritdoc IGovernance
-    function withdrawLQTY(uint88 _lqtyAmount) external nonReentrant {
+    function withdrawLQTY(uint256 _lqtyAmount) external nonReentrant {
         // check that user has reset before changing lqty balance
         UserState storage userState = userStates[msg.sender];
         require(userState.allocatedLQTY == 0, "Governance: must-allocate-zero");
 
         UserProxy userProxy = UserProxy(payable(deriveUserProxyAddress(msg.sender)));
         require(address(userProxy).code.length != 0, "Governance: user-proxy-not-deployed");
 
-        uint88 lqtyStaked = uint88(stakingV1.stakes(address(userProxy)));
+         // Update the offset tracker
+        if (_lqtyAmount < userState.unallocatedLQTY) {
```

Comment:
Are we checking that the input parameter is <= `unallocatedLQTY`? Otherwise it would revert below when trying to subtract.
Maybe we can update `userState.unallocatedLQTY` inside the `if` clause, zeroing it in the `else` branch, so we would be effectively clamping the input amount to the max.

Reply:
Right, but if they try to withdraw more than their unallocatedLQTY, isn't revert the right behavior?  But I'm fine to clamp to the max too